sched/clustermembership/dynamicCluster.go

```diff
@@ -1,49 +1,77 @@
 package cluster_membership
 
-/*
- * Represents a Cluster of nodes that can change over time.
- * Keeps a Map of the nodeId to Node for fast lookup of nodes.
- */
+//
+// Represents a Cluster of nodes that can change over time.
+// Keeps a Map of the nodeId to Node for fast lookup of nodes.
+//
 type dynamicCluster struct {
 	memberSet   map[string]bool
 	membersList []Node
+	updateCh    chan NodeUpdate
```

Comment:
Maybe have it be a slice of channels, and then iterate over them? Then you could have multiple watchers, and not have to have also the watcher variable?